src/main/java/com/squareup/catalog/demo/example/clone/CloneCatalogExample.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright 2017 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.catalog.demo.example.clone;
+
+import com.squareup.catalog.demo.example.Example;
+import com.squareup.connect.ApiClient;
+import com.squareup.connect.ApiException;
+import com.squareup.connect.api.CatalogApi;
+import com.squareup.connect.api.LocationsApi;
+import com.squareup.connect.auth.OAuth;
+import com.squareup.connect.models.BatchUpsertCatalogObjectsRequest;
+import com.squareup.connect.models.BatchUpsertCatalogObjectsResponse;
+import com.squareup.connect.models.CatalogObject;
+import com.squareup.connect.models.CatalogObjectBatch;
+import com.squareup.connect.models.ListCatalogResponse;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.UUID;
+
+/**
+ * This example clones catalog objects from one merchant account to another. The accessToken
+ * specified when the script is executed is assumed to be the source account from which catalog
+ * objects are copied.
+ *
+ * NOTE: This example is in development. Currently it only clones discounts, modifier lists, and
+ * taxes. Items are not yet cloned.
+ */
+public class CloneCatalogExample extends Example {
+
+  public CloneCatalogExample() {
+    super("clone_catalog", "Clones catalog objects from one merchant account to another.");
+  }
+
+  @Override
+  public void execute(CatalogApi sourceCatalogApi, LocationsApi locationsApi) throws ApiException {
+    // Prompt the user for configuration info.
+    Config config = promptForConfigurationOptions();
+
+    // Create a CatalogApi to access the target account.
+    CatalogApi targetCatalogApi =
+        createTargetCatalogApi(sourceCatalogApi, config.targetAccessToken);
+
+    try {
+       /*
```

Comment:
Yes, but it adds complexity because of the memberships. I figured I'd put this out there because it's useful, then follow-up with support for items.